miner/worker.go

```diff
@@ -288,12 +288,19 @@ func (w *worker) start() {
 	w.startCh <- struct{}{}
 
 	if istanbul, ok := w.engine.(consensus.Istanbul); ok {
-		istanbul.SetBlockProcessors(w.chain.HasBadBlock,
+		istanbul.SetCallBacks(w.chain.HasBadBlock,
 			func(block *types.Block, state *state.StateDB) (types.Receipts, []*types.Log, uint64, error) {
 				return w.chain.Processor().Process(block, state, *w.chain.GetVMConfig())
 			},
-			func(block *types.Block, state *state.StateDB, receipts types.Receipts, usedGas uint64) error {
-				return w.chain.Validator().ValidateState(block, state, receipts, usedGas)
+			w.chain.Validator().ValidateState,
+			func(block *types.Block, receipts []*types.Receipt, logs []*types.Log, state *state.StateDB) {
+				_, err := w.chain.WriteBlockWithState(block, receipts, logs, state, true)
+				if err != nil {
+					log.Error("Failed writeBlockWithState in onNewConsensusBlock", "blockNumber", block.Number(), "hash", block.Hash(), "err", err)
+					return
+				}
+				log.Info("Successfully imported new block in onNewConsensusBlock", "number", block.Number(), "hash", block.Hash(), "elapsed")
+				w.mux.Post(core.NewMinedBlockEvent{Block: block})
```

Comment:
this also returns an error, and we are silently ignoring it here. 

We cant' do much here, but at least we should log it